### PR TITLE
[client][change-capture] Tweak client factory to rely on venice store config

### PR DESCRIPTION
**Pull request by ZacAttack**
At some point we'd like to have the ability to transmit configurations to the view clients based on view configurations within the store itself. This also prevents users from spinning up clients to stores which haven't been properly configured yet.

This feature isn't used yet so no client impact expected.

## Summary, imperative, start upper case, don't end with a period

Resolves #XXX

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerClientFactory.java [hunk 1]
```diff
@@ -32,11 +39,32 @@ public synchronized <K, V> VeniceChangelogConsumer<K, V> getChangelogConsumer(St
...
           .setSchemaReader(ClientFactory.getSchemaReader(newStoreChangelogClientConfig.getInnerClientConfig()));
-      if (newStoreChangelogClientConfig.getViewClassName()
-          .equals(ChangeCaptureView.CHANGE_CAPTURE_VIEW_WRITER_CLASS_NAME)) {
+
+      String viewClass = getViewClass(
+          storeName,
+          newStoreChangelogClientConfig.getViewClassName(),
+          d2ControllerClient,
+          globalChangelogClientConfig.getControllerRequestRetryCount());
+      if (viewClass.equals(ChangeCaptureView.class.getCanonicalName())) {
```
Comment (haoxu07) [inline internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerClientFactory.java:53]: Minor: as we created the LOGGER int this class, maybe we can log here?  or we should not? (as there are some logs inside the changelog class constructor)

File: internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerClientFactory.java [hunk 2]
```diff
@@ -40,11 +40,16 @@ public synchronized <K, V> VeniceChangelogConsumer<K, V> getChangelogConsumer(St
...
-          newStoreChangelogClientConfig.getViewClassName(),
-          d2ControllerClient,
-          globalChangelogClientConfig.getControllerRequestRetryCount());
+      String viewClass = newStoreChangelogClientConfig.getViewClassName() == null
+          ? ""
+          : newStoreChangelogClientConfig.getViewClassName();
+      if (!viewClass.isEmpty()) {
+        viewClass = getViewClass(
+            storeName,
+            newStoreChangelogClientConfig.getViewClassName(),
```
Comment (haoxu07) [inline internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerClientFactory.java:47]: Just another minor thing: we might need to change config field name: `newStoreChangelogClientConfig.getViewClassName()` to `newStoreChangelogClientConfig.getViewName()`. It is not a class name anymore?

**Review by haoxu07 — APPROVED** [review]
LGTM! Seems there are some test coverage issues.

**Review by sixpluszero — DISMISSED** [review]
Looks good but it fails at test coverage, which honestly I don't think would be super easy to add if we don't game the coverage test (like testing against setViewClassName which I think is not super useful....but since you have integration added, maybe simple unit test should be fine.)

**Review by sixpluszero — APPROVED** [review]
LGTM, thanks!

**Pull request merged**